### PR TITLE
making clang-tidy work properly again

### PR DESCRIPTION
**Pull request by sithhell**
This PR makes sure that the code is properly checked with the below enabled checks:
   
     - All of bugprone except:
        - bugprone-assert-side-effect
        - bugprone-exception-escape
        - bugprone-forward-declaration-namespace
        - bugprone-macro-parentheses
     - Removing removed tests

This PR includes the fixes that appeared.

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -3,11 +3,19 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-Checks: '-*,modernize-use-nullptr,misc-use-after-move,misc-virtual-near-miss,misc-multiple-statement-macro,misc-move-constructor-init,misc-move-forwarding-reference,misc-assert-side-effect,misc-dangling-handle,misc-non-copyable-objects,misc-forwarding-reference-overload,misc-unused-raii'
```
Comment (msimberg) [inline .clang-tidy]: Are all of the `misc-*` checks (except for `misc-non-copyable-objects`) part of `bugprone-*` now?

File: tests/unit/util/any.cpp [hunk 2]
```diff
@@ -132,7 +132,7 @@ int hpx_main(variables_map& vm)
             HPX_TEST(!any1.empty());
             any any2(std::move(any1));
             HPX_TEST(!any2.empty());
-            HPX_TEST(any1.empty());
+            HPX_TEST(any1.empty()); // NOLINT
```
Comment (msimberg) [inline tests/unit/util/any.cpp]: What warnings did this cause?

File: src/util/runtime_configuration.cpp [hunk 3]
```diff
@@ -133,7 +133,7 @@ namespace hpx { namespace util
             "[hpx]",
             "location = ${HPX_LOCATION:$[system.prefix]}",
             "component_paths = ${HPX_COMPONENT_PATHS}",
-            "component_base_paths = $[hpx.location]"
+            "component_base_paths = $[hpx.location]" // NOLINT
```
Comment (msimberg) [inline src/util/runtime_configuration.cpp]: What warning did this cause?

File: .clang-tidy (hunk 1 above)
Comment (sithhell) [reply .clang-tidy]: yes

File: tests/unit/util/any.cpp (hunk 2 above)
Comment (sithhell) [reply tests/unit/util/any.cpp]: use after move, which this code is explicitly testing for.

File: src/util/runtime_configuration.cpp (hunk 3 above)
Comment (sithhell) [reply src/util/runtime_configuration.cpp]: use after move, which this code is explicitly testing for.

Comment (msimberg) [reply src/util/runtime_configuration.cpp]: The one in `tests/unit/util/any.cpp` I get, but this? No test here...

Comment (sithhell) [reply src/util/runtime_configuration.cpp]: grr, sorry. actually readin where you commented helps ;)
This was:
`/home/heller/programming/hpx/src/util/runtime_configuration.cpp:136:13: error: suspicious string literal, probably missing a comma [bugprone-suspicious-missing-comma,-warnings-as-errors]`

**Pull request merged**